pkg/startup/maxprocs.go

```diff
@@ -0,0 +1,21 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package startup
+
+import (
+	"log"
+	"runtime"
+)
+
+// LimitMaxProcs limits the GOMAXPROCS to the number specified.
+func LimitMaxProcs(maxProcs int) {
+	curProcs := runtime.GOMAXPROCS(0)
+
+	if curProcs > maxProcs {
+		runtime.GOMAXPROCS(maxProcs)
+
+		log.Printf("limited GOMAXPROCS to %d", maxProcs)
```

Comment:
should we also pass in the service name?

Reply:
it is printed to the service log actually, so it should be clear which service is that, e.g. `talosctl logs apid`